ui/cypress/integration/common/volume.js

```diff
@@ -11,68 +11,98 @@ And('I choose the Volumes tag', () => {
+  const volumeLabelName = dataTableObject.labelName;
+  const volumeLabelValue = dataTableObject.labelValue;
+
+  cy.get('input[name=name]').type(volumeName);
+  cy.get('input[name=labelName]').type(volumeLabelName);
+  cy.get('input[name=labelValue]').type(volumeLabelValue);
+  cy.get('[data-cy=add-volume-labels-button]').click();
+
+  // node name
+  cy.get('.sc-select').eq(0).click();
+  cy.get('.sc-select__menu').eq(0).click();
+
+  cy.get('.sc-select').eq(1).click();
+  cy.get('.sc-select__menu').find('[data-cy=storageClass-metalk8s]').click();
+
+  cy.get('.sc-select').eq(2).click();
+  cy.get('.sc-select__menu').find(`[data-cy="type-${volumeType}"]`).click();
+
+  cy.get('input[name=sizeInput]').type(volumeSize);
 });
 
-Then(
-  'I fill out the create volume form with SparseLoopDevice volume type and check if the status is Ready',
-  function () {
-    cy.route('GET', '/api/kubernetes/api/v1/persistentvolumes').as(
-      'getPersistentVolumes',
-    );
-    const volumeNameSparseLoopDevice = `volume-${new Date().getTime()}`;
-    const volumeCapacity = Cypress.env('volume_capacity');
-    const volume_label_name = Cypress.env('volume_label_name');
-    const volume_label_value = Cypress.env('volume_label_value');
-
-    cy.get('input[name=name]').type(volumeNameSparseLoopDevice);
-
-    cy.get('input[name=labelName]').type(volume_label_name);
-    cy.get('input[name=labelValue]').type(volume_label_value);
-    cy.get('[data-cy=add-volume-labels-button]').click();
-
-    // node name
-    cy.get('.sc-select').eq(0).click();
-    cy.get('.sc-select__menu').eq(0).click();
-
-    cy.get('.sc-select').eq(1).click();
-    cy.get('.sc-select__menu').find('[data-cy=storageClass-metalk8s]').click();
-
-    cy.get('.sc-select').eq(2).click();
-    cy.get('.sc-select__menu')
-      .find('[data-cy="type-sparseLoopDevice"]')
-      .click();
-
-    cy.get('input[name=sizeInput]').type(volumeCapacity);
-
-    cy.get('.sc-select').eq(2).click();
-    //cy.get('[data-cy="size-KiB"]').click(); leave it at the default value GiB
-
-    cy.get('[data-cy="submit-create-volume"]').click();
-
-    // Wait until the volume is ready
-    cy.waitUntil(
-      () =>
-        cy
-          .get('[data-cy=volume_status_value]')
-          .then(($span) => $span.text() === 'Ready'),
-      {
-        errorMsg: `Volume ${volumeNameSparseLoopDevice} is not ready`,
-        timeout: 120000, // waits up to 120000 ms, default to 5000
-        interval: 5000, // performs the check every 5000 ms, default to 200
-      },
-    );
-
-    // Check if all the labels are presents
-    cy.get('[data-cy=volume_label_name]').should('contain', volume_label_name);
-    cy.get('[data-cy=volume_label_value]').should(
-      'contain',
-      volume_label_value,
-    );
-  },
-);
+And('I click [Create] button', () => {
+  cy.get('[data-cy="submit-create-volume"]').click();
+});
+
+Then('I am redirected to the "test-volume-sparse" volume page', () => {
```

Comment:
This one would ideally allow arbitrary names (so it's reusable for other tests). [Reference](https://github.com/TheBrainFamily/cypress-cucumber-preprocessor#reusable-step-definitions)

```suggestion
Then('I am redirected to the "{string}" volume page', (volumeName) => {
```

Reply:
Definitely yes!!!